main.tf

```diff
@@ -52,3 +45,17 @@ resource "snowflake_user" "this" {
   rsa_public_key   = local.rsa_public_key
   rsa_public_key_2 = var.rsa_public_key_2
 }
+
+resource "snowflake_role_grants" "default_role" {
+  count = var.grant_default_roles && var.default_role != null ? 1 : 0
```

Comment:
shouldn't we check also `module.this.enabled`?

Reply:
Good catch - added